src/services/entity_builder.js

```diff
@@ -41,47 +35,51 @@ export default class EntityBuilder {
   }
 
   validateAsset(asset) {
-    validatePathsNotEmpty(asset, [
-      'assetId',
-      'content.idData',
-      'content.signature',
-      'content.idData.createdBy',
-      'content.idData.timestamp',
-      'content.idData.sequenceNumber'
-    ]);
-    this.ensureTimestampWithinLimit(asset.content.idData.timestamp);
-    validateFieldsConstrainedToSet(asset, ['content', 'assetId']);
-    validateFieldsConstrainedToSet(asset.content, ['idData', 'signature']);
-
-    this.identityManager.validateSignature(asset.content.idData.createdBy, asset.content.signature, asset.content.idData);
+    validateAndCast(asset)
+      .required([
+        'assetId',
+        'content.idData',
+        'content.signature',
+        'content.idData.createdBy',
+        'content.idData.timestamp',
+        'content.idData.sequenceNumber'
+      ])
+      .fieldsConstrainedToSet(['content', 'assetId'])
+      .fieldsConstrainedToSet(['idData', 'signature'], 'content')
+      .validate(['content.idData.timestamp'], this.isTimestampWithinLimit.bind(this),
```

Comment:
I find `bind` particularly ugly, and would prefer an arrow function that calls `this.isTimestamp` instead.